dali/operators/reader/loader/numpy_loader.cc

```diff
@@ -38,6 +37,115 @@ TypeInfo TypeFromNumpyStr(const std::string &format) {
   DALI_FAIL("Unknown Numpy type string");
 }
 
+inline void SkipSpace(const char*& ptr) {
+  while(::isspace(*ptr))
+    ptr++;
+}
+
+template <size_t N>
+void Skip(const char*& ptr, const char (&what)[N]) {
+  DALI_ENFORCE(!strncmp(ptr, what, N-1), make_string("Expected \"", what, "\" but got \"", ptr, "\""));
+  ptr += N - 1;
+}
+
+template <size_t N>
+bool TrySkip(const char*& ptr, const char (&what)[N]) {
+  if (!strncmp(ptr, what, N-1)) {
+    ptr += N - 1;
+    return true;
+  } else {
+    return false;
+  }
+}
+
+
+template <size_t N>
+void SkipFieldName(const char*& ptr, const char (&name)[N]) {
+  SkipSpace(ptr);
+  Skip(ptr, "'");
+  Skip(ptr, name);
+  Skip(ptr, "'");
+  SkipSpace(ptr);
+  Skip(ptr, ":");
+  SkipSpace(ptr);
+}
+
+std::string ParseStringValue(const char*& input, char delim_start = '\'', char delim_end = '\'') {
+  DALI_ENFORCE(*input++ == delim_start, make_string("Expected \'", delim_start, "\'"));
+  std::string out;
```

Comment:
I think you should use `stringstream` here. E. g. `out+='\\'` creates a new object from scratch, is that right?

Reply:
Good point.
How += behaves, is implementation defined (haha). Some of behave like vector, so += is efficient, others don't. Since we've recently found out that we're using quite dated libstdc++

Reply:
Once I had this very argument, where I was on the side of using std::stringstream and the other person saying that += is as good if not better. Measurement showing += to be the winner.